### ISSUE TITLE
[UiAutomator2] Problem. Can't run tests

### ISSUE THREAD

**Issue by S-Borkowski** [appium/appium#12331]
## The problem

For the past few days i tried to run tests with appium, plot twist, it didn't work. On my previous project UiAutomator(1) did run well. I tried everything, please help :<

## Environment

* Appium v1.11.1
* Ubuntu 18.04
* Node.js v8.15.1
* Npm
* Nexus_6_API_28
* Emulator
* Appium CLI 
Edit. Same on Appium Desktop

## Details

I can't run UiAutomator 2. Tried running: appium, sudo appium and sudo -E appium.

## Link to Appium logs

https://gist.github.com/S-Borkowski/05c40fa94c40a2ea0f706ca184386c21

## Code To Reproduce Issue
```
    public void setUpBrowser() {
        dc = new DesiredCapabilities();
        dc.setCapability("deviceName",
                "Android_SDK_built_for_x86");
        //dc.setCapability("app",
        //        "/resources/base.apk");
        dc.setCapability("platformName",
                "Android");
        dc.setCapability("automationName",
                "UiAutomator2");
        dc.setCapability("appPackage",
                "com.company.project");
    }

    public void setUpMethod() {
        dc.setCapability("appActivity",
                "com.company.project." + activity);
        try {
            driver = new AndroidDriver(new URL("http://0.0.0.0:4723/wd/hub"), dc);
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        softAssert = new SoftAssert();
    }
```

**Comment by dpgraham** [comment]
The gist link is broken

**Comment by S-Borkowski** [comment]
> The gist link is broken

should work now, sorry

**Comment by dpgraham** [comment]
Can you run logcat and provide the logs for around when the error happens?

**Comment by S-Borkowski** [comment]
Sure, if you can walk me through it.

**Comment by dpgraham** [comment]
Just type `adb logcat` from your command line. Then run the test, and copy and paste the contents of the logcat around when the crash happens.

**Comment by S-Borkowski** [comment]
https://gist.github.com/S-Borkowski/67e6c270586314e4bf32b9838acbc4bc

**Comment by mykola-mokhnach** [comment]
Are you sure the logcat is taken within the correct time range? I don't observe any messages from instrumentation or UIA2 server there

**Comment by S-Borkowski** [comment]
100% sure but let me try again

**Comment by S-Borkowski** [comment]
https://gist.github.com/S-Borkowski/67e6c270586314e4bf32b9838acbc4bc

**Comment by mykola-mokhnach** [comment]
> 03-18 12:17:09.039 19658 19683 E AndroidRuntime: java.lang.IllegalStateException: UiAutomationService android.accessibilityservice.IAccessibilityServiceClient$Stub$Proxy@1eedf06already registered!

This looks like some internal UiAutomation framework issue. Try to reset the device under test or replace it with another one.

**Comment by S-Borkowski** [comment]
I'm using android studio and build in android emulator. Which device do you recommend me to emulate?

**Comment by mykola-mokhnach** [comment]
Try to completely delete the emulator and create it again. Also, you could try appium@beta.

**Comment by S-Borkowski** [comment]
1. Ok
2. Sure, if you give me command to paste into terminal

**Comment by mykola-mokhnach** [comment]
https://appiumpro.com/editions/47

**Comment by S-Borkowski** [comment]
Same thing on v1.11.0-beta.3 and new device :/    
Appsigner error something in appium logs

**Comment by mykola-mokhnach** [comment]
No logs - no solution

**Comment by S-Borkowski** [comment]
appium beta - https://gist.github.com/S-Borkowski/4e6b9e4ed03f4e092775b6cf11705fa7    
sudo appium beta - https://gist.github.com/S-Borkowski/a75eccffb29cb27c3ef91ec1182da56d    
adb logcat on appium beta - https://gist.github.com/S-Borkowski/74304b7763600f7981b11b26637b5d89    
adb logcat on sudo appium beta - closed immediately
sudo -E appium beta - https://gist.github.com/S-Borkowski/1c0e5f32c0f5e804c79498ce4a907bf0    
This one finally worked. You have some problems with permissions, no one can install you app with just npm install -g appium, even with sudo there are problems. Maybe linux gods can install and use your app without problems, but mere mortals like myself practically can't :/

**Comment by mykola-mokhnach** [comment]
Using sudo was never a good idea. People say that installing appium with `--unsafe-perms` enabled might do the trick or simply giving R/W perms to `/usr/local/lib/node_modules/appium/node_modules/appium-uiautomator2-server/apks` folder

**Comment by S-Borkowski** [comment]
You could update your installation guide ;)

**Comment by mykola-mokhnach** [comment]
Marked as documentation issue. Feel free to improve the existing Appium documentation with the troubleshooting solution if you have time

**Comment by john-kheir** [comment]
Had the same exact issue on  Ubuntu 18.04. Installing appium  `--unsafe-perms` definitely did the trick
